### ISSUE TITLE
[BUG/ISSUE] gchp crashed on error "MPIR_Get_contextid_sparse_group(600): Too many communicators" after about 13days of simulatoin

### ISSUE THREAD

**Issue by hgrhgy** [geoschem/gchp#187]
The GCHP crashes after about 13 days of simulation. I tried it several times and each time it crashed at about 13 days simulation. The rundir is created default configuration.
It feels like a configuration issue, but I don't know which one to configure?

env:
> GCHP version: 13.3.3
> gcc: 9.4.0
> gfortran: 9.4.0
> esmf: 8.1.1
> mpich: 3.3.2

```log 
GENERIC: INFO: Started the  'Run' stage of the gridded component 'EXTDATA'
Fatal error in PMPI_Win_allocate_shared: Other MPI error, error stack:
PMPI_Win_allocate_shared(163).......: MPI_Win_allocate_shared(size=0, MPI_INFO_NULL, comm=0xc40033e9, baseptr=0x7fffc0b12e20, win=0x7fffc0b12e70) failed
MPID_Win_allocate_shared(208).......: 
win_init(276).......................: 
MPIR_Comm_dup_impl(57)..............: 
MPII_Comm_copy(571).................: 
MPIR_Get_contextid_sparse_group(600): Too many communicators (8/2048 free on this process; ignore_id=0)
Fatal error in PMPI_Win_allocate_shared: Other MPI error, error stack:
PMPI_Win_allocate_shared(163).......: MPI_Win_allocate_shared(size=0, MPI_INFO_NULL, comm=0xc40033e6, baseptr=0x7ffffa643020, win=0x7ffffa643070) failed
MPID_Win_allocate_shared(208).......: 
win_init(276).......................: 
MPIR_Comm_dup_impl(57)..............: 
MPII_Comm_copy(571).................: 
MPIR_Get_contextid_sparse_group(600): Too many communicators (8/2048 free on this process; ignore_id=0)
```

[gchp.log](https://github.com/geoschem/GCHP/files/7658769/gchp.log)
[runConfig.sh.txt](https://github.com/geoschem/GCHP/files/7658771/runConfig.sh.txt)
[gchp.local.run.txt](https://github.com/geoschem/GCHP/files/7658775/gchp.local.run.txt)

**Comment by lizziel** [comment]
Hi @hgrhgy, this looks like an error of too many MPI communicators open at once. Given it runs for 13 days before encountering this I wonder if there is a communicator leak. Would you be able to try using OpenMPI rather than MPICH, or a newer version of MPICH, to see if that makes a difference? Those should have higher communicator limits.

**Comment by hgrhgy** [comment]
I have update the mpich 3.3.2 to 3.4.2, the problem still exists. Is there any sugestion to debug this problem?

**Comment by mathomp4** [comment]
Not a solution, but maybe MPICH 4 doesn't have this issue? This thread maybe says it might get around it:

https://lists.mpich.org/pipermail/discuss/2021-October/006259.html

but it's hard to say. Probably more an issue in MAPL that Open MPI and Intel MPI let us "get past".

**Comment by lizziel** [comment]
I create a MAPL issue for this: https://github.com/GEOS-ESM/MAPL/issues/1232.

**Comment by lizziel** [comment]
This issue was resolved by a patch provided by GMAO [here](https://github.com/GEOS-ESM/MAPL/commit/4f43d87455b90e0aa83f3fd33f1b5083af3ad72f). The fix had already been incorporated into a more recent version of MAPL than was is used in GCHP 13.0-13.3. We will bring it into 13.4.

**Comment by LiamBindle** [comment]
Reopened and self assigned to myself. I'll close this when I merge the fix from GMAO.

**Comment by LiamBindle** [comment]
Hi @huangyj and @lizziel, #191 upgrades MAPL to 2.16. According to https://github.com/GEOS-ESM/MAPL/issues/1232 this was fixed in 2.6.3, so this should be fixed in GCHP 13.4.0.

**Comment by LiamBindle** [comment]
The MAPL 2.16 upgrade is postponed to 14.0

**Comment by kilicomu** [comment]
I have read through the MAPL issue and found the communicator freeing patch (which I will apply to my copy of the model code), just wanted to +1 this with Intel MPI 2018.5 and GCHP 13.2.1.

**Comment by lizziel** [comment]
I will bring the patch for this fix into 13.4.1.
